### PR TITLE
Fix missing sensor and rebalance issues

### PR DESCRIPTION
**Pull request by patkivikram**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://faust.readthedocs.io/en/master/contributing.html).

## Description

Fix all the race conditions in rebalance and recovery

### CODE REVIEW

**Review by bobh66 — APPROVED** [review]
Looks good to me

File: faust/stores/rocksdb.py [hunk 1]
```diff
@@ -377,26 +385,40 @@ async def assign_partitions(self, table: CollectionT, tps: Set[TP]) -> None:
...
                 # side effect: opens db and adds to self._dbs.
+                self.logger.info(
+                    f"opening partition {partition} for gen id "
+                    f"{generation_id} app id {self.app.consumer_generation_id}"
+                )
                 return self._db_for_partition(partition)
             except rocksdb.errors.RocksIOError as exc:
                 if i == max_retries - 1 or "lock" not in repr(exc):
                     # release all the locks and crash
+                    self.log.warning(
```
Comment (jkgenser) [inline faust/stores/rocksdb.py:409]: Why is it `self.log` here but `self.logger` above?

**Review by jkgenser — APPROVED** [review]
I can confirm that scale up and scale down with 0 standby replicas and one shared data dir works with this setup. 

It also requires `faust-streaming-rocskdb==0.8.0`. Probably will experience rocksdb db locks if you try the shared data dir approach with `python-rocksdb==0.7.0`

**Comment by patkivikram** [comment]
Also implements https://github.com/faust-streaming/faust/issues/98

**Pull request merged**